File: enum-jpa-validation-app/src/main/java/com/example/enumapp/common/code/CodeEnum.java
package com.example.enumapp.common.code;

/**
 * DB/API 코드값을 가지는 Enum 추상화.
 * MyBatis TypeHandler 와 Jackson Converter 의 공통 대상이다.
 */
public interface CodeEnum {
    String getCode();
    String getDescription();
}

File: enum-jpa-querydsl-kotlin-app/src/main/kotlin/com/example/enumapp/common/code/CodeEnum.kt
package com.example.enumapp.common.code

/**
 * DB/API 코드값을 가지는 Enum 추상화.
 * MyBatis TypeHandler 와 Jackson Converter 의 공통 대상이다.
 */
interface CodeEnum {
    val code: String
    val description: String
}
